src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1309,7 +1309,6 @@ private IOperation CreateBoundCollectionInitializerCollectionLiteralExpression(B
             }
             else
             {
-                // PROTOTYPE: Temporary until natural type is supported.
```

Comment:
I'm assuming we now enter the previous 'if' in the natural type scenario?

Reply:
Yes, that's correct.